Jasmine/Common/ViewControllers/BaseGameViewControllerDelegate.swift

```diff
@@ -0,0 +1,31 @@
+import Foundation
+
+/// Sets the shared functionalities that are applicable across all game view controller delegates.
+/// This protocol will be inherited by all the specialsied game view controller delegates.
+protocol BaseGameViewControllerDelegate {
+
+    // MARK: Score Update
+    /// Redisplay the score displayed on the view controller screen with a new score.
+    ///
+    /// - Parameter newScore: the new score to be redisplayed.
+    func redisplay(_ newScore: Int)
+
+    // MARK: Time Update
+    /// Redisplay the time remaining on the view controller against a total time.
+    ///
+    /// - Parameters:
+    ///   - timeRemaining: the remaining time left, in seconds.
+    ///   - totalTime: the total time from the start of the game, in seconds.
+    ///     If no bonus time is added, should equate to `initialRemainingTime`.
+    func redisplay(_ timeRemaining: TimeInterval, outOf totalTime: TimeInterval)
```

Comment:
same as above. `redisplay(timeRemaining: 5, outOf: 10)` would be clearer than `redisplay(5, outOf: 10)` since we are dealing with redisplaying score etc as well.